packages/amplify-migration-codegen-e2e/src/assertions.ts

```diff
@@ -5,50 +5,108 @@ import {
   getFunction,
   getAppSyncApi,
   describeCloudFormationStack,
+  getUserPoolClients,
 } from '@aws-amplify/amplify-e2e-core';
 import { getProjectOutputs } from './projectOutputs';
 import { getAppSyncDataSource, getResourceDetails } from './sdk_calls';
 import { removeProperties } from '.';
+import { $TSAny } from '@aws-amplify/amplify-cli-core';
 
-export async function assertGen1Setup(projRoot: string) {
+async function getMeta(projRoot: string) {
   const gen1Meta = getProjectMeta(projRoot);
   const gen1Region = gen1Meta.providers.awscloudformation.Region;
-  const { UserPoolId: gen1UserPoolId } = Object.keys(gen1Meta.auth).map((key) => gen1Meta.auth[key])[0].output;
-  const { Arn: gen1FunctionArn, Name: gen1FunctionName } = Object.keys(gen1Meta.function).map((key) => gen1Meta.function[key])[0].output;
-  const { BucketName: gen1BucketName } = Object.keys(gen1Meta.storage).map((key) => gen1Meta.storage[key])[0].output;
-  const {
-    GraphQLAPIIdOutput: gen1GraphqlApiId,
-    GraphQLAPIEndpointOutput,
-    GraphQLAPIKeyOutput,
-  } = Object.keys(gen1Meta.api).map((key) => gen1Meta.api[key])[0].output;
-  const { graphqlApi } = await getAppSyncApi(gen1GraphqlApiId, gen1Region);
-
-  expect(gen1Region).toBeDefined();
+  return { gen1Meta, gen1Region };
+}
 
+async function assertUserPool(gen1Meta: $TSAny, gen1Region: string) {
```

Comment:
can we also assert the userpool client has been code-genned correctly ?

Reply:
You mean, compare the resource properties of gen1 and gen2 Userpool client resource using cloud control api?

Reply:
yup similar to what we do for userpool

Reply:
Yes, this helps and I was able to catch more bugs. Thanks.

Reply:
Addressed. [dbf44b4](https://github.com/aws-amplify/amplify-cli/pull/13978/commits/dbf44b417f56692be3608d684dc8f31f985b6df8)